TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/XDrive.java

```diff
@@ -0,0 +1,267 @@
+package org.firstinspires.ftc.teamcode.robot.subsystems;
+
+import static org.firstinspires.ftc.teamcode.robot.constants.XDriveConstants.*;
+
+import androidx.annotation.NonNull;
+
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+import com.qualcomm.robotcore.hardware.IMU;
+
+
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.YawPitchRollAngles;
+
+import org.firstinspires.ftc.teamcode.core.lib.autonomousControl.MotorVelocityData;
+import org.firstinspires.ftc.teamcode.core.lib.autonomousControl.Pose2d;
+import org.firstinspires.ftc.teamcode.core.lib.autonomousControl.RobotMovementState;
+import org.firstinspires.ftc.teamcode.core.lib.autonomousControl.TargetVelocityData;
+import org.firstinspires.ftc.teamcode.core.lib.gamepad.GamepadManager;
+import org.firstinspires.ftc.teamcode.core.lib.gamepad.SmartGamepad;
+import org.firstinspires.ftc.teamcode.core.lib.interfaces.Subsystem;
+import org.firstinspires.ftc.teamcode.core.lib.pid.PIDController;
+import org.firstinspires.ftc.teamcode.robot.constants.AutonomousConstants;
+import org.firstinspires.ftc.teamcode.robot.constants.DrivetrainBuilderConstants;
+import org.firstinspires.ftc.teamcode.robot.constants.XDriveConstants;
+import org.opencv.core.Mat;
+
+public class XDrive implements Subsystem {
+    enum DrivetrainState{
+        MANUAL_CONTROL,
+        APRIL_TAG_ALIGNMENT
+    }
+    public DrivetrainState currentDriveState = DrivetrainState.MANUAL_CONTROL;
+    private  Telemetry telemetry;
+    private static XDrive instance;
+    private double reset_angle = 0;
+    private DcMotor front_left = null;
+    private DcMotor back_left = null;
+    private DcMotor back_right = null;
+    private DcMotor front_right = null;
+    private IMU imu;
+
+
+    private PIDController XpositionPID=AutonomousConstants.stopAtPointPID;
+    private PIDController YpositionPID=AutonomousConstants.stopAtPointPID;
+    private PIDController HpositionPID=AutonomousConstants.stopAtPointPID;
+
+    Pose2d currentPose = START_POSITION;
+    int prevFLTicks1 = 0;
```

Comment:
add private +
you can have like this:
int prevFLTicks1 = 0, prevFRTicks1 = 0,  [...] ,  prevBRTicks2 = 0;

Reply:
done